include/layers/Tensor.hpp

```diff
@@ -0,0 +1,166 @@
+#pragma once
+
+#include <cstdint>
+#include <iostream>
+#include <stdexcept>
+#include <type_traits>
+#include <vector>
+
+#include "layers/Shape.hpp"
+
+enum class Type { kUnknown, kInt, kDouble };
+
+template <typename T>
+std::vector<uint8_t>* to_byte(std::vector<T>& v) {
+  return reinterpret_cast<std::vector<uint8_t>*>(&v);
+}
+
+template <typename T>
+const std::vector<uint8_t>* to_byte(const std::vector<T>& v) {
+  return reinterpret_cast<const std::vector<uint8_t>*>(&v);
+}
+
+template <typename T>
+Type GetTypeEnum() {
+  if constexpr (std::is_same<T, int>::value) {
+    return Type::kInt;
+  } else if constexpr (std::is_same<T, double>::value) {
+    return Type::kDouble;
+  } else {
+    return Type::kUnknown;
+  }
+}
+
+class Tensor {
+ private:
+  Shape shape_;
+  std::vector<uint8_t> values_;
+  Type type_;
+
+  std::vector<uint8_t> SetRightTypeValues() {
+    if (type_ == Type::kInt) {
+      return std::vector<uint8_t>(shape_.count() * sizeof(int), 0);
+    }
+    if (type_ == Type::kDouble) {
+      return std::vector<uint8_t>(shape_.count() * sizeof(double), 0);
+    }
+    return std::vector<uint8_t>();
+  }
+
+  template <typename T>
+  std::vector<T>* as();
+
+  template <typename T>
+  const std::vector<T>* as() const;
+
+ public:
+  Tensor(const std::vector<uint8_t>& a, const Shape& s, Type type) {
+    type_ = type;
+    shape_ = s;
+    values_ = SetRightTypeValues();
+
+    if (a.size() != values_.size() || type == Type::kUnknown) {
+      throw std::invalid_argument("Wrong_Length");
+    }
+    values_ = a;
+  }
+
+  Tensor(const std::vector<size_t>& dims, Type type) : shape_(dims) {
+    type_ = type;
+    values_ = SetRightTypeValues();
+  }
+
+  Tensor(const Shape& sh, Type type) : shape_(sh) {
+    type_ = type;
+    values_ = SetRightTypeValues();
+  }
+
+  Tensor(const Tensor& t) = default;
+
+  Tensor(Tensor&& t) = default;
+
+  Tensor& operator=(Tensor&& t) = default;
+
+  std::vector<uint8_t> get_values() const { return values_; }
```

Comment:
For what do we need this function? I think we need to work with tensor without copy memory (exclude get one element)

Reply:
I'll add operator+ then, because i need to sum vector at some point

Reply:
@NeiroYT tensor is atomic object, all arithmetic operation with tensor need to be work outside tensor or I don't understand why we need operator+

Reply:
I use std::transform to sum vector (tensor) with bias (tensor). Or should I do it without std::transform?

Reply:
Ok i'll make sum function outside of the tensor while refactoring

Reply:
@NeiroYT 
created https://github.com/embedded-dev-research/itlab_2023/issues/90